Subject: [PATCH] handle quoted module parameters
X-BK-ChangeSetKey: rddunlap@osdl.org[torvalds]|ChangeSet|20050105052605|56827
Date:	Wed, 05 Jan 2005 11:05:19 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2072, 2005/01/04 21:26:05-08:00, rddunlap@osdl.org

	[PATCH] handle quoted module parameters
	
	Fix module parameter quote handling.
	Module parameter strings (with spaces) are quoted like so:
	"modprm=this test"
	and not like this:
	modprm="this test"
	
	Signed-off-by: Randy Dunlap <rddunlap@osdl.org>
	Signed-off-by: Andrew Morton <akpm@osdl.org>
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 params.c |   15 ++++++++++++---
 1 files changed, 12 insertions(+), 3 deletions(-)


diff -Nru a/kernel/params.c b/kernel/params.c
--- a/kernel/params.c	2005-01-04 23:21:24 -08:00
+++ b/kernel/params.c	2005-01-04 23:21:24 -08:00
@@ -77,10 +77,16 @@
 static char *next_arg(char *args, char **param, char **val)
 {
 	unsigned int i, equals = 0;
-	int in_quote = 0;
+	int in_quote = 0, quoted = 0;
+	char *next;
 
 	/* Chew any extra spaces */
 	while (*args == ' ') args++;
+	if (*args == '"') {
+		args++;
+		in_quote = 1;
+		quoted = 1;
+	}
 
 	for (i = 0; args[i]; i++) {
 		if (args[i] == ' ' && !in_quote)
@@ -106,13 +112,16 @@
 			if (args[i-1] == '"')
 				args[i-1] = '\0';
 		}
+		if (quoted && args[i-1] == '"')
+			args[i-1] = '\0';
 	}
 
 	if (args[i]) {
 		args[i] = '\0';
-		return args + i + 1;
+		next = args + i + 1;
 	} else
-		return args + i;
+		next = args + i;
+	return next;
 }
 
 /* Args looks like "foo=bar,bar2 baz=fuz wiz". */
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

